File: mobile/lib/user_display.dart
import 'roles.dart';

/// Etiqueta principal en chat/radio (indicativo: SGTO GOMEZ, B.O. LINARES…).
String userDisplayLabel(Map<String, dynamic>? user, {String fallback = 'Usuario'}) {
  final name = user?['displayName']?.toString().trim();
  if (name != null && name.isNotEmpty) return name;
  final username = user?['username']?.toString().trim();
  if (username != null && username.isNotEmpty) return username;
  return fallback;
}

/// Subtítulo: usuario de login · rol (ej. aacunap2 · Operador).
String userDisplaySubtitle(Map<String, dynamic>? user) {
  final parts = <String>[];
  final username = user?['username']?.toString().trim();
  if (username != null && username.isNotEmpty) parts.add(username);
  final role = user?['role']?.toString();
  if (role != null && role.isNotEmpty) {
    parts.add(roleLabel(role));
  }
  return parts.join(' · ');
}

File: web/src/userDisplay.js
/**
 * Etiqueta visible en chat/radio (indicativo: SGTO GOMEZ, B.O. LINARES…).
 * Usa displayName de BD (callSign); fallback a username.
 */
export function userDisplayLabel(user, fallback = 'Usuario') {
  const name = String(user?.displayName || '').trim();
  if (name) return name;
  const username = String(user?.username || '').trim();
  if (username) return username;
  return fallback;
}

/** Subtítulo: usuario de login · rol */
export function userDisplaySubtitle(user) {
  const parts = [];
  const username = String(user?.username || '').trim();
  if (username) parts.push(username);
  const role = String(user?.role || '').trim();
  if (role) parts.push(role);
  return parts.join(' · ');
}
